File: Leetcode/203-Remove-Linked-List-Elements/RemoveLinkedListElements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    if (!head) {
        return head
    }
    var dummy = new ListNode(0)
    dummy.next = head
    var cur = dummy
    while (cur) {
        var ite = cur.next
        var exists = false
        while (ite && ite.val == val) {
            exists = true
            ite = ite.next
        }
        if (exists) {
            cur.next = ite
        }
        cur = ite
    }
    return dummy.next
};

File: Leetcode/203-Remove-Linked-List-Elements/RemoveLinkedListElements.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
	if head == nil {
		return head
	}
	dummy := new(ListNode)
	dummy.Next = head
	cur := dummy
	for cur != nil {
		ite := cur.Next
		exists := false
		for ite != nil && ite.Val == val {
			exists = true
			ite = ite.Next
		}
		if exists {
			cur.Next = ite
		}
		cur = ite
	}
	return dummy.Next
}
